src/frontend/src/features/providers/mailbox.tsx

```diff
@@ -180,6 +189,50 @@ export const MailboxProvider = ({ children }: PropsWithChildren) => {
         }
     });
 
+    const threadEventsQuery = useThreadsEventsList(
+        selectedThread?.id ?? '',
+        undefined,
+        {
+            query: {
+                enabled: !!selectedThread,
+            }
+        }
+    );
+
+    // Merge messages and events into a single timeline sorted by created_at
+    const timelineItems = useMemo<TimelineItem[] | null>(() => {
+        if (!messagesQuery.data?.data?.results || !selectedThread) return null;
+
+        const items: TimelineItem[] = [];
+
+        // Add messages
+        messagesQuery.data.data.results.forEach((message) => {
+            items.push({
+                type: 'message',
+                data: message,
+                created_at: message.created_at,
+            });
+        });
+
+        // Add events
+        if (threadEventsQuery.data?.data?.results) {
+            threadEventsQuery.data.data.results.forEach((event) => {
+                items.push({
+                    type: 'event',
+                    data: event,
+                    created_at: event.created_at,
+                });
+            });
+        }
+
+        // Sort by created_at
+        return items.sort((a, b) => {
+            const dateA = new Date(a.created_at).getTime();
+            const dateB = new Date(b.created_at).getTime();
+            return dateA - dateB;
+        });
+    }, [messagesQuery.data?.data?.results, threadEventsQuery.data?.data?.results, selectedThread]);
+
```

Comment:
Oh... by reading that I realize both message and thread event endpoints are paginated but we don't use an InfiniteQuery to fetch those resources so currently we always fetch only the first page...
Can we imagine to remove pagination from those resources (Is it acceptable from a performance point of view ?) ? It will ease the logic to merge data and ease its display.

Reply:
up to you. It would make sense once it's more mature to merge all that in a single api call for a thread. I guess there will be huge threads so paginating would be a plus even if we put the default at ~100 messages.